controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -306,70 +324,74 @@ private ApplicationController applications() {
     /**
      * Finds the next step to trigger for the given application, if any, and returns these as a list.
      */
-    private List<Job> computeReadyJobs(ApplicationId id) {
+    private List<Job> computeReadyJobs(TenantAndApplicationId id) {
         List<Job> jobs = new ArrayList<>();
-        applications().get(id).ifPresent(application -> {
-            Change change = application.change();
-            Optional<Instant> completedAt = max(application.deploymentJobs().statusOf(systemTest)
-                                                        .<Instant>flatMap(job -> job.lastSuccess().map(JobRun::at)),
-                                                application.deploymentJobs().statusOf(stagingTest)
-                                                        .<Instant>flatMap(job -> job.lastSuccess().map(JobRun::at)));
-            String reason = "New change available";
-            List<Job> testJobs = null; // null means "uninitialised", while empty means "don't run any jobs".
-            DeploymentSteps steps = steps(application.deploymentSpec());
-
-            if (change.hasTargets()) {
-                for (Step step : steps.production()) {
-                    List<JobType> stepJobs = steps.toJobs(step);
-                    List<JobType> remainingJobs = stepJobs.stream().filter(job -> ! isComplete(change, application, job)).collect(toList());
-                    if ( ! remainingJobs.isEmpty()) { // Change is incomplete; trigger remaining jobs if ready, or their test jobs if untested.
-                        for (JobType job : remainingJobs) {
-                            Versions versions = Versions.from(change, application, deploymentFor(application, job),
-                                                              controller.systemVersion());
-                            if (isTested(application, versions)) {
-                                if (completedAt.isPresent() && canTrigger(job, versions, application, stepJobs)) {
-                                    jobs.add(deploymentJob(application, versions, change, job, reason, completedAt.get()));
+        applications().getApplication(id).ifPresent(application -> {
+            for (Instance instance : application.instances().values()) {
```

Comment:
Perhaps this should be only the default instance, until properly supported?

Reply:
I don't think it matters. We'll do the instance orchestration very soon, and I don't think anyone is using more than one production instance yet.